backend/ops_api/ops/resources/agreements.py

```diff
@@ -247,22 +234,38 @@ def patch(self, id: int) -> Response:
 class AgreementListAPI(BaseListAPI):
     def __init__(self, model: BaseModel = Agreement):
         super().__init__(model)
-        self._post_schema_contract = desert.schema(ContractAgreementRequestBody)
-        self._post_schema_grant = desert.schema(GrantAgreementRequestBody)
-        self._get_schema = desert.schema(QueryParameters)
+        self._schema_contract = desert.schema(ContractAgreementData)
+        self._schema_grant = desert.schema(GrantAgreementData)
 
     @staticmethod
-    def _get_query(search=None, research_project_id=None):
-        stmt = select(Agreement).order_by(Agreement.id)
+    def _get_query(args):
+        polymorphic_agreement = with_polymorphic(Agreement, [ContractAgreement, GrantAgreement])
```

Comment:
Yes. That did it all. We just add the other subclasses to the list part, and it will assign correctly.

Reply:
The "with_polymorphic" does a OUTER JOIN on all those tables, and then allows for you to have all the fields for searchability. The problem is that when looking up a specific field, I still needed to do the specific table name, which is why there's the if/elif/else pattern.

Reply:
Yes, I could do a similar thing as the if...elif...else pattern in the code with a match, however it doesn't look as clean in this instance, because I am checkin if a value is in a set of values, and while it is somewhat clean, I don't find it as elegant or readable as an if...elif...else block.